### PR TITLE
[Tizen] Run single thread for processing all GLib sources

### PR DESCRIPTION
**Pull request by arkq**
### Problem

Currently Tizen platform spawns a new thread every time an async Tizen API call is about to be made. This is required to make sure that Tizen API internally will attach IO sources to correct context - context which is processed by Matter event loop. A dedicated Matter event loop is required, because it is possible, that Tizen application will not run default GLib main event loop.

This PR is a first step for adding proper matter stack locking when processing GLib-based Tizen events.

### Changes

- run Matter GLib main event loop in a dedicated thread started during Matter stack initialization
- synchronically invoke all async Tizen API calls on the Matter GLib context (so it will be processed by our dedicated thread)
- removed MainLoop class

### Testing

Tested locally by performing BLE commissioning Linux-Tizen, Tizen-Linux and Tizen-Tizen.
CI will test network commissioning.

### CODE REVIEW

File: src/platform/Tizen/BLEManagerImpl.h [hunk 1]
```diff
@@ -207,8 +207,8 @@ class BLEManagerImpl final : public BLEManager,
     void NotifyBLEWriteReceived(System::PacketBufferHandle & buf, BLE_CONNECTION_OBJECT conId);
 
     // ==== Connection.
-    void ConnectHandler(const char * address);
-    static gboolean ConnectChipThing(gpointer userData);
+    // void ConnectHandler(const char * address);
```
[the remark this answers is not in the document]
Comment (arkq) [reply src/platform/Tizen/BLEManagerImpl.h]: Done

**Review by dh79pyun — APPROVED** [review]
Looks good

**Comment by hyunuktak** [comment]
It looks good overall. and i have a question for you.
In the changed structure, is there an action for timeout if the call is not completed for the asynchronous function call?

**Comment by arkq** [comment]
> In the changed structure, is there an action for timeout if the call is not completed for the asynchronous function call?

@hyunuktak , no, currently there is no such concept as timeout for `GLibMatterContextInvokeSync()` calls. This function is not async, but synchronous by design. The purpose of this function is to run given callback on GLib context, that's all. The next step for this refactoring (in another PR) will be proper locking of matter stack when running callback function. However, we are not sure whether it will be possible without duplicating glib internal functions. So, other possibility is that the dedicated glib thread will be removed. So, for internal calls the `g_main_context_invoke_full()` in the `GLibMatterContextInvokeSync()` will not switch threads at all - callback will be called right away. If one wants async processing it will have to be done separately, e.g. use glib timeout source (but not with `g_idle_add()` - this will use global main context, and we do not want that).

**Review by andy31415 — APPROVED** [review]

**Pull request merged**